### PR TITLE
Add a configuration property to ban ClusterFormationTasks

### PR DESCRIPTION
**Pull request by alpar-t**
The intent is tofold:

- first, make sure what we think is converted actually is
- second, lather on, make sure that some checks in CI don't these tasks,
  so that once everything is converted we can be sure no new tests are
  added that follow the old ways

### CODE REVIEW

**Comment by mark-vieira** [comment]
This seems a bit unnecessary to be honest. We'll also need to go and toggle this on all the respective CI builds for which the legacy method is "banned" until everything is fully migrated. I don't think we risk new usages of this leaking into the codebase given one of us reviews all build changes.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy [hunk 1]
```diff
@@ -55,6 +55,9 @@ class ClusterFormationTasks {
      * Returns a list of NodeInfo objects for each node in the cluster.
      */
     static List<NodeInfo> setup(Project project, String prefix, Task runner, ClusterConfiguration config) {
+        if ("false".equals(System.getProperty("tests.allow.ClusterFormationTasks", "true"))) {
+            throw new IllegalStateException("Using cluster formation tasks but these are not allowed in this build");
```
Comment (mark-vieira) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy]: If we do this, we should give folks better instruction. Even so far as to say "Ping Alpar on Slack for more info" :wink:

At the very least we should point folks towards `TestClusters`.

**Review by mark-vieira — CHANGES_REQUESTED** [review]
To be honest, this seems a bit unnecessary. I don't think we risk much new usages of this leaking into the build given we review all build changes. All the needed configuration on the Jenkins side is just another thing that we'll need to go back and remove once all this is fully migrated and removed entirely.

**Comment by rjernst** [comment]
I agree with @mark-vieira, this looks like overkill.

**Pull request closed without merging**